core/camera/src/main/java/com/google/jetpackcamera/core/camera/DebugCameraInfoUtil.kt

```diff
@@ -0,0 +1,113 @@
+        val result = JSONArray()
+        for (logicalCameraId in cameraManager.cameraIdList) {
+            val logicalCameraData = JSONObject()
+            logicalCameraData.put(
+                "logical-$logicalCameraId",
+                getCameraPropertiesJSONObject(logicalCameraId, cameraManager)
+            )
+            for (
+            physicalCameraId in
+            cameraManager.getCameraCharacteristics(logicalCameraId).physicalCameraIds
+            ) {
+                logicalCameraData.put(
+                    "physical-$physicalCameraId",
+                    getCameraPropertiesJSONObject(physicalCameraId, cameraManager)
+                )
+            }
+            result.put(logicalCameraData)
+        }
+        return result
+    }
+
+    private fun getCameraPropertiesJSONObject(
+        cameraId: String,
+        cameraManager: CameraManager
+    ): JSONObject {
+        val cameraCharacteristics =
+            cameraManager.getCameraCharacteristics(cameraId)
+        val jsonObject = JSONObject()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            cameraCharacteristics.get(CameraCharacteristics.LENS_POSE_ROTATION)
+                ?.let { jsonObject.put(CameraCharacteristics.LENS_POSE_ROTATION.name, it) }
+            cameraCharacteristics.get(CameraCharacteristics.LENS_POSE_TRANSLATION)
+                ?.let { jsonObject.put(CameraCharacteristics.LENS_POSE_TRANSLATION.name, it) }
+            cameraCharacteristics.get(CameraCharacteristics.LENS_INTRINSIC_CALIBRATION)
+                ?.let { jsonObject.put(CameraCharacteristics.LENS_INTRINSIC_CALIBRATION.name, it) }
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            cameraCharacteristics.get(CameraCharacteristics.LENS_DISTORTION)
+                ?.let { jsonObject.put(CameraCharacteristics.LENS_DISTORTION.name, it) }
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            cameraCharacteristics.get(CameraCharacteristics.CONTROL_ZOOM_RATIO_RANGE)
+                ?.let { jsonObject.put(CameraCharacteristics.CONTROL_ZOOM_RATIO_RANGE.name, it) }
+        }
+        cameraCharacteristics.get(CameraCharacteristics.LENS_INFO_AVAILABLE_FOCAL_LENGTHS)
+            ?.let {
+                jsonObject.put(
+                    CameraCharacteristics.LENS_INFO_AVAILABLE_FOCAL_LENGTHS.name,
+                    it
+                )
+            }
+        cameraCharacteristics.get(CameraCharacteristics.LENS_INFO_MINIMUM_FOCUS_DISTANCE)
+            ?.let {
+                jsonObject.put(
+                    CameraCharacteristics.LENS_INFO_MINIMUM_FOCUS_DISTANCE.name,
+                    it
+                )
+            }
+        cameraCharacteristics.get(CameraCharacteristics.REQUEST_AVAILABLE_CAPABILITIES)
+            ?.let { jsonObject.put(CameraCharacteristics.REQUEST_AVAILABLE_CAPABILITIES.name, it) }
+
+        return jsonObject
```

Comment:
After trying this out, many of these are types like `FloatArray` which don't have a good `toString()` implementation. You should check the return type of each of these `CameraCharacteristics` and use functions like `FloatArray?.contentToString()` to write them to the file.

Reply:
good catch!